ROADMAP.md

```diff
@@ -0,0 +1,587 @@
+environments with libp2p greatly expands libp2p's userbase on the one side and
+on the other enables existing libp2p deployments to interconnect with said IoT
+devices.
+
+### 🎓 libp2p as a platform for Networks Research & Innovation
+
+**What?** Academics should be able to use libp2p as a playground for
+their research. Amongst other points, this encompasses:
+
+1.  Building a public utility p2p network that academics can deploy
+    their algorithms and solutions onto, to trial them on a large
+    scale, or test their hypothesis on.
+
+2.  *Packaging P2P prior art in composable building blocks,* to allow
+    researchers to focus on the problem at hand, without having to
+    reinvent the wheel.
+
+3.  Providing modelling tools, data collection hooks, sampling
+    strategies, etc. to facilitate research practices.
+
+4.  Making libp2p stable and dependable enough so researchers are
+    satisfied with the *rigor* of studies built on libp2p.
+
+This goal has parallels with the TestLab endeavour.
+
+**Why?** To contribute back to the p2p research community; to harness
+the work taking place at research facilities by implementing the
+research outputs on libp2p.
+
+### 🚑 Self-healing networks
+
+**What?** A number of nodes in the network take on important assistive
+roles, e.g. bootstrap, DHT boosters, relay, network, etc. However:
+
+1. relying on static IPs or DNS entries exposes us to centralisation,
+   and is fragile.
+
+2. the network grows and shrinks dynamically, so we should scale these
+   nodes accordingly.
+
+3. these nodes are easy to attack directly, and such attacks can harm
+   the network; currently we have no resilience plan.
+
+4. routing/connectivity between nodes could be compromised by network
+   attacks, or censorship.
+
+In the long term, we aim to build a resilient, self-healing network that
+scales dynamically based in response to traffic volume, disasters, and
+network conditions.
+
+**Why?** To get closer to the dream of unstoppable, intelligent,
+adaptive networks.
+
+### 📮 Offline message queue / postbox
+
+**What?** Providing solutions to store data for nodes that have gone
+offline temporarily. Payloads can be pubsub messages, DHT provider
+records, or even RPC calls. In traditional MQ systems, this property is
+typically termed "reliability".
+
+Several ideas have been floated within the libp2p community:
+
+- Third-party postbox nodes that are incentivised to buffer data between nodes A
+  and B, when either is unavailable.
+
+- Store-and-forward architectures.
+
+And there's probably many more to explore. When reasoning about these
+methods, we should keep in mind aspects like replication, storage,
+liveness, byzantine fault tolerance, etc. Depending on the mechanism,
+the challenges may even align with those of Filecoin.
+
+**Why?** It's an enabler for offline-first scenarios, low quality connections,
+roaming, etc.
+
+**Links:**
+
+-   https://github.com/libp2p/notes/issues/2
```

Comment:
Did some experiments with performant authenticated replicated streams this week. Achieving speeds of 220MB/s was fairly easy, but I think this could be pushed even further. Not sure if this is of interest here.

- https://github.com/dvc94ch/blake-streams
- https://github.com/ipfs-rust/ipfs-embed/pull/81